### PR TITLE
Feat/HNG-177: Implement Custom Textarea with Validation and Character Count

### LINKED ISSUE #177: [FEATURE] 🗺️ ⚙️ Settings Pages > Profile Settings > implement custom textarea with validation and character count
### Description

Develop a Textarea with Character Limit component. This component will allow users to input text while displaying the character count and providing an error message when the character limit is exceeded. The component should be user-friendly, visually appealing, and easy to integrate into various forms.

### **Acceptance Criteria**

The Textarea with Character Limit component is implemented according to the design.
The component should display the current character count.
Provide an error message when the character limit is exceeded.

**Requirements**
The component should receive the following props:
maxLength: The maximum number of characters allowed.
value: The current value of the textarea.
onChange: A callback function that is triggered when the textarea value changes.

### **Testing**
Ensure the component correctly tracks and displays the character count.
Check that the component provides an error message when the character limit is exceeded.
Test the component for responsiveness and accessibility.
Verify that the onChange callback is triggered with the correct value.

textarea (1)

### Acceptance Criteria

Tasks
- [ ] Develop the Textarea with Character Limit component based on the provided requirements.
- [ ] Implement handling for tracking and displaying the character count.
- [ ] Ensure real-time feedback and error message for exceeding the character limit.
- [ ] Write unit tests to ensure the component works as expected.
- [ ] Test the component for responsiveness and compatibility across different  devices.

### Links

https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=10-16395&t=ZQNRsvC3jgSF9JiA-1

### Images

_No response_

### PR DESCRIPTION
**Pull request by JoshuaAkpan**
# Description

**Closes #177  **

# Changes proposed

## What were you told to do?

Develop a Textarea with Character Limit component.

## What did you do?

1. Created the CharacterLimitTextarea component.
- Props include maxLength, value, onChange, id, name, and label.
- Tracks character count and handles errors when the character limit is exceeded.
- Displays the character count and error messages conditionally.

2. Implemented the useCharacterLimit custom hook.
- Manages character count and error state.
- Updates state based on the current value and maxLength.

3. Added unit tests in CharacterLimitTextarea.test.tsx.
- Tests for rendering, character count updates, and error message display.
- Ensures onChange callback is correctly triggered.

4. Added index.tsx for easy import of the CharacterLimitTextarea component.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [X] My code follows the code style of this project.
- [X] This PR does not contain plagiarized content.
- [X] The title and description of the PR is clear and explains the approach.
- [X] I am making a pull request against the **dev branch** (left side).
- [X] My commit messages styles matches our requested structure.
- [X] My code additions will fail neither code linting checks nor unit test.
- [X] I am only making changes to files I was requested to.

# Screenshots/Videos

Initial State

Character Count Updating

Error Message

### CODE REVIEW

File: html/favicon.svg
Comment (prudentbird) [inline html/favicon.svg:1]: Why is this file changed?

File: html/index.html
Comment (prudentbird) [inline html/index.html:1]: Why is this file changed ?

File: src/app/page.tsx
Comment (prudentbird) [inline src/app/page.tsx:1]: Why is this file changed?

File: src/test/CharacterLimitTextarea.test.tsx
Comment (prudentbird) [inline src/test/CharacterLimitTextarea.test.tsx:1]: Use userevent over fireevent in your tests

**Review by prudentbird — COMMENTED** [review]
Fix your merge conflicts. Pull upstream and do the needful!

File: src/test/CharacterLimitTextarea.test.tsx
Comment (JoshuaAkpan) [reply src/test/CharacterLimitTextarea.test.tsx:1]: As instructed, I have implemented the change to use userEvent over fireEvent in my tests.

File: src/app/page.tsx
Comment (JoshuaAkpan) [reply src/app/page.tsx:1]: That was because I had finished my work before the styles.md was added which indicated that we could preview our work in http://localhost:3000/guides {That is the pages.tsx file in it}. Upon seeing the updates, I had to return page.tsx to default.

File: html/index.html
Comment (JoshuaAkpan) [reply html/index.html:1]: I didn't tamper with it. When my server loads, that just refreshes. So it's included when I commit

File: html/favicon.svg
Comment (JoshuaAkpan) [reply html/favicon.svg:1]: I didn't tamper with it. When my server loads, that just refreshes. So it's included when I commit

**Comment by oreoluwa212** [comment]
Update your branch to match dev and make a PR asap please (). ensure lint and test passes
